src/InputGroup.js

```diff
@@ -0,0 +1,37 @@
+import styled from 'styled-components'
+import system, { space, borderColor, propTypes } from 'styled-system'
+import theme from './theme'
+import Box from './Box'
+import Input from './Input'
+
+const InputGroup = styled.div`
+  display: flex;
+  align-items: center;
+  border-radius: ${system.theme('radius')};
+  border-width: 1px;
+  border-style: solid;
+  ${borderColor}
+  ${space}
+
+  & > ${Box} {
+    width: 100%;
+    flex: 1 1 auto;
+  }
+
+  & ${Input} {
```

Comment:
+1 on this "advanced" example of style selector syntax in `styled-components` with a design system component.

Reply:
Normally, I'd say avoid nested selectors like this, but I think the InputGroup is an exception to that rule

Reply:
I like this!